inmap/cmd/inmap.go

```diff
@@ -64,17 +65,34 @@ var DefaultScienceFuncs = []inmap.CellManipulator{
 // runtime, and cleanup, respectively.
 func Run(dynamic, createGrid bool, scienceFuncs []inmap.CellManipulator, addInit, addRun, addCleanup []inmap.DomainManipulator) error {
 
+	startTime := time.Now()
+
 	// Start a function to receive and print log messages.
+	logfile, err := os.Create(Config.Logfile)
+	if err != nil {
+		log.Fatal("Cannot create file", err)
+	}
+	defer logfile.Close()
+	mw := io.MultiWriter(os.Stdout, logfile)
+	log.SetOutput(mw)
 	cConverge := make(chan inmap.ConvergenceStatus)
 	cLog := make(chan *inmap.SimulationStatus)
 	msgLog := make(chan string)
+	var walltime float64
 	go func() {
 		for {
 			select {
 			case msg := <-cConverge:
-				fmt.Println(msg.String())
+				log.Println(msg.String())
+				maxChange := msg[0]
+				for _, v := range msg {
```

Comment:
fmt.Println prints to the console but not to the log. log.Println prints to the console and to the log.